### PR TITLE
black and white win loss ratio

### PR DESCRIPTION
**Pull request by EllAchE**

### CODE REVIEW

File: src/metrics/misc.ts [hunk 1]
```diff
@@ -174,16 +164,60 @@ export class MetadataMetric implements Metric {
...
+    this.mostGamesPlayed = maxGames;
+    this.playerMostGames = playerMostGames;
+
+    return {
+      averagePlayerRating: this.averagePlayerRating,
+      averageRatingDiff: this.averageRatingDiff,
+      mostGamesPlayed: maxGames,
+      playerMostGames,
+      // This doesn't account for ties
+      whiteWinRatio: this.whiteWins / (this.whiteWins + this.blackWins),
```
Comment (bennyrubanov) [inline src/metrics/misc.ts:220]: `whiteWinRatioWithTies: (this.whiteWins + (this.ties/2)) / (this.whiteWins + this.blackWins + (this.ties / 2))`

to calculate the ratio of white wins to black wins including counting a tie as half a point for both black and white

**Review by bennyrubanov — APPROVED** [review]

File: src/metrics/misc.ts (hunk 1 above)
Comment (EllAchE) [reply src/metrics/misc.ts:220]: didn't know how we wanted to calc exaclty, introducing any of these is ez

Comment (EllAchE) [reply src/metrics/misc.ts:220]: merging & we can add l8r

File: src/metrics/misc.ts [hunk 2]
```diff
@@ -174,16 +164,60 @@ export class MetadataMetric implements Metric {
...
+
+    if (metadata[4] === '[Result "1-0"]') {
+      this.whiteWins++;
+    } else if (metadata[4] === '[Result "0-1"]') {
+      this.blackWins++;
+    } else if (metadata[4] === '[Result "1/2-1/2"]') {
+      this.ties++;
+    } else {
+      throw new Error('Invalid result');
     }
```
Comment (bennyrubanov) [inline src/metrics/misc.ts:185]: tested this code, apparently PGN result is not always at index 4. going to change this in latest PR

**Pull request merged**